cmd/apiKeyValueMapUpload_test.go

```diff
@@ -0,0 +1,68 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRunApiKeyValueMapUpload(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfull Api Key Value Map Create Test", func(t *testing.T) {
+		// init
+		apiServiceKey := `{
+			"oauth": {
+				"url": "https://demo",
+				"clientid": "demouser",
+				"clientsecret": "******",
+				"tokenurl": "https://demo/oauth/token"
+			}
+		}`
+		config := apiKeyValueMapUploadOptions{
+			APIServiceKey:   apiServiceKey,
+			Key:             "demo",
+			Value:           "name",
+			KeyValueMapName: "demoMap",
+		}
+		httpClient := httpMockCpis{CPIFunction: "ApiKeyValueMapUpload", ResponseBody: ``, TestType: "PositiveCase"}
+		// test
+		err := runApiKeyValueMapUpload(&config, nil, &httpClient)
+		// assert
+		if assert.NoError(t, err) {
+			t.Run("check url", func(t *testing.T) {
+				assert.Equal(t, "https://demo/apiportal/api/1.0/Management.svc/KeyMapEntries", httpClient.URL)
+			})
+
+			t.Run("check method", func(t *testing.T) {
+				assert.Equal(t, "POST", httpClient.Method)
+			})
+		}
+	})
+
+	t.Run("Failed case of API Key Value Map Create Test", func(t *testing.T) {
```

Comment:
Additional negative tests:
- apiProxyUploadStatusResp.StatusCode != http.StatusCreated
- apiProxyUploadStatusResp == nil
- challenge: make sure apiProxyUploadStatusResp.Body.Close() was closed
- "json payload is invalid for key value map"

Reply:
added some new tests